src/test/java/com/ning/http/client/providers/grizzly/NonBlockingInputStreamFeederTest.java

```diff
@@ -67,6 +72,31 @@ public void whenWriteQueuIsFullTheOnQueueWriteStrategyIsAppliedAndAllTheDataIsCo
     assertFeeding(DATA);
   }
 
+  @Test
+  public void whenExecutionExceptionIsThrownWhileQueueIsBlocking_thenDoNotWaitForQueueToBeFree() throws Exception {
```

Comment:
I ran this test on my local, and changed the code to return always true in NonBlockingInputStreamFeeder (just to mimic the previous behavior). It reproduces the infinite loop, but because this line is thowing a NullPointerException:
```final long writeTimeout = c.getTransport().getWriteTimeout(MILLISECONDS);```
(in FeedableBodyGenerator#BaseFeeder.block).

The change is fine, but can you mock that line in this test so that it reflects the reported infinite loop?

It would be just:
```
Transport transport = mock(Transport.class);
when(connection.getTransport()).thenReturn(transport);
when(transport.getWriteTimeout(any(TimeUnit.class))).thenReturn( /* here you can return -1 or a positive value to end up calling get() or get(with timeout)
```